paddle/majel/allocation.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <memory>
+
+#include <majel/place.h>
+#include "hl_gpu.h"
```

Comment:
Also, please try to include header files in .cc/cu files, other than in .h file, because the inclusion of `hl_gpu.h` propagates to all .cc/.cu files that include `allocation.h`.